File: 0190-reverse-bits/0190-reverse-bits.cpp
class Solution {
public:
    uint32_t reverseBits(uint32_t n) {
        uint32_t mask = 0;
        for (int i = 0; i < 32; i++) {
            if (n & (1 << i)) {
                mask |= (1 << (31 - i));
            }
        }
        return mask;
    }
};

File: 0190-reverse-bits/0190-reverse-bits.py
class Solution:
    def reverseBits(self, n: int) -> int:
        mask = (1 << 32) - 1
        res = 0
        for i in range(32):
            if n & (1 << i):
                res = res | (1 << (31 - i))
                # res &= mask
        return res